libs/sdk-ui-dashboard/src/model/store/catalog/catalogSelectors.ts

```diff
@@ -340,3 +388,42 @@ function getAttributesWithHierarchyDescendants(
     });
     return attributeDescendants;
 }
+
+function buildAdhocDateHierarchies(
+    dateHierarchyTemplates: IDateHierarchyTemplate[],
+    catalogDateDatasets: ICatalogDateDataset[],
+): ICatalogAttributeHierarchy[] {
+    const adhocDateHierarchies: ICatalogAttributeHierarchy[] = [];
+    catalogDateDatasets.forEach((dateDataset) => {
+        const dateAttributes = dateDataset.dateAttributes;
+        const defaultDateHierarchyTemplate = dateHierarchyTemplates.find(
+            (template) => template.id === "default",
```

Comment:
IMHo this is not necessary. If we will introduce date templates metadata objects, only thing that must be changed is SPI implementation. The rest of the code should work out of box. This condition ties it only to the harcoded template. All the returned templates by SPI should be processed.

Reply:
Do you mean that if we have multiple date templates, we will create multiple hierarchies for each date dataset?

Reply:
fixed, handled all templates returned by SPI